js/adapt-contrib-pageLevelProgress.js

```diff
@@ -1,66 +1,135 @@
 define([
     'core/js/adapt',
     './completionCalculations',
-    './PageLevelProgressMenuView',
-    './PageLevelProgressNavigationView'
-], function(Adapt, completionCalculations, PageLevelProgressMenuView, PageLevelProgressNavigationView) {
-
-    function setupPageLevelProgress(pageModel, enabledProgressComponents) {
-        new PageLevelProgressNavigationView({model: pageModel, collection: new Backbone.Collection(enabledProgressComponents)});
-    }
-
-    // This should add/update progress on menuView
-    Adapt.on('menuView:postRender', function(view) {
-        if (view.model.get('_id') == Adapt.location._currentId) return;
-
-        var coursePLPConfig = Adapt.course.get('_pageLevelProgress');
-
-        // do not proceed if pageLevelProgress is not enabled in course.json
-        if (!coursePLPConfig || !coursePLPConfig._isEnabled) {
-            return;
+    './PageLevelProgressNavigationView',
+    './PageLevelProgressIndicatorView',
+    './PageLevelProgressCollection'
+], function(Adapt, completionCalculations, PageLevelProgressNavigationView, PageLevelProgressIndicatorView, PageLevelProgressCollection) {
+
+    var PageLevelProgress = Backbone.Controller.extend({
+
+        initialize: function() {
+            this.listenTo(Adapt, 'app:dataReady', this.onDataReady);
+        },
+
+        getCourseConfig: function() {
+            return Adapt.course.get('_pageLevelProgress');
+        },
+
+        onDataReady: function() {
+            // Do not proceed if pageLevelProgress is not enabled in course.json
+            var coursePLPConfig = this.getCourseConfig();
+            if (!coursePLPConfig || !coursePLPConfig._isEnabled) {
+                return;
+            }
+            this.setUpEventListeners();
+        },
+
+        setUpEventListeners: function() {
+            var headerIndicatorTypes = [
+                'menu',
+                'page',
+                'article',
+                'block',
+                'component'
+            ];
```

Comment:
Not really related to this, but perhaps the framework should expose such an array?

Reply:
It'd be pretty easy to derive from the models. I'm quite sure Daryl once said he didn't want to hard-code them in the framework like this. I can't remember why. I'm inclined to agree that at a certain level the relationships should be a tad more abstract but when we define the view / model for article / block etc that hard-coding is fine.
Still, if this array is derivable, that covers us in the future if we decide to add any more model types.

Reply:
https://github.com/adaptlearning/adapt_framework/issues/2285 added issue